lib/src/base.dart

```diff
@@ -511,6 +522,474 @@ class Event {
+          }
+        }
+      }
+    }
+
+    return json;
+  }
+}
+
+/// This describes the device that caused the event.
+class Device {
+  /// The name of the device. This is typically a hostname.
+  final String name;
+
+  /// The family of the device.
+  ///
+  /// This is normally the common part of model names across generations.
+  /// For instance `iPhone` would be a reasonable family,
+  /// so would be `Samsung Galaxy`.
+  final String family;
+
+  /// The model name. This for instance can be `Samsung Galaxy S3`.
+  final String model;
+
+  /// An internal hardware revision to identify the device exactly.
+  final String modelId;
+
+  /// The CPU architecture.
+  final String arch;
+
+  /// If the device has a battery, this can be an floating point value
+  /// defining the battery level (in the range 0-100).
+  final double batteryLevel;
+
+  /// Defines the orientation of a device.
+  final Orientation orientation;
+
+  /// The manufacturer of the device.
+  final String manufacturer;
+
+  /// The brand of the device.
+  final String brand;
+
+  /// The screen resolution. (e.g.: `800x600`, `3040x1444`).
+  final String screenResolution;
+
+  /// A floating point denoting the screen density.
+  final String screenDensity;
+
+  /// A decimal value reflecting the DPI (dots-per-inch) density.
+  final String screenDpi;
+
+  /// Whether the device was online or not.
+  final bool online;
+
+  /// Whether the device was charging or not.
+  final bool charging;
+
+  /// Whether the device was low on memory.
+  final bool lowMemory;
+
+  /// A flag indicating whether this device is a simulator or an actual device.
+  final bool simulator;
+
+  /// Total system memory available in bytes.
+  final int memorySize;
+
+  /// Free system memory in bytes.
+  final int freeMemory;
+
+  /// Memory usable for the app in bytes.
+  final int usableMemory;
+
+  /// Total device storage in bytes.
+  final int storageSize;
+
+  /// Free device storage in bytes.
+  final int freeStorage;
+
+  /// Total size of an attached external storage in bytes
+  /// (e.g.: android SDK card).
+  final int externalStorageSize;
+
+  /// Free size of an attached external storage in bytes
+  /// (e.g.: android SDK card).
+  final int externalFreeStorage;
+
+  /// When the system was booted
+  final DateTime bootTime;
+
+  /// The timezone of the device, e.g.: `Europe/Vienna`.
+  final String timezone;
+
+  const Device({
+    this.name,
+    this.family,
+    this.model,
+    this.modelId,
+    this.arch,
+    this.batteryLevel,
+    this.orientation,
+    this.manufacturer,
+    this.brand,
+    this.screenResolution,
+    this.screenDensity,
+    this.screenDpi,
+    this.online,
+    this.charging,
+    this.lowMemory,
+    this.simulator,
+    this.memorySize,
+    this.freeMemory,
+    this.usableMemory,
+    this.storageSize,
+    this.freeStorage,
+    this.externalStorageSize,
+    this.externalFreeStorage,
+    this.bootTime,
```

Comment:
We recently added some new fields which I Don't think made it to the docs yet. Not at the computer right now but I can check later and make a PR.